br/pkg/storage/s3.go

```diff
@@ -416,12 +416,18 @@ func (rs *S3Storage) DeleteFile(ctx context.Context, file string) error {
 
 // FileExists check if file exists on s3 storage.
 func (rs *S3Storage) FileExists(ctx context.Context, file string) (bool, error) {
-	input := &s3.HeadObjectInput{
-		Bucket: aws.String(rs.options.Bucket),
-		Key:    aws.String(rs.options.Prefix + file),
+	var err error
+	key := aws.String(rs.options.Prefix + file)
+	// empty path means check the bucket itself
+	if len(*key) == 0 || *key == "/" {
+		err = checkS3Bucket(rs.svc, rs.options)
+	} else {
+		input := &s3.HeadObjectInput{
+			Bucket: aws.String(rs.options.Bucket),
+			Key:    key,
+		}
+		_, err = rs.svc.HeadObjectWithContext(ctx, input)
```

Comment:
Update the check logic by do a `WalkDir`. Because s3/gcs does not support check non-regular file like directory, I leave the `FileExist` method to only support check regular file as its original logic.